perf_test/blas/blas/KokkosBlas_trtri_perf_test.hpp

```diff
@@ -0,0 +1,424 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 3.0
+//       Copyright (2020) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY NTESS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL NTESS OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Siva Rajamanickam (srajama@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#ifndef KOKKOSBLAS_TRTRI_PERF_TEST_H_
+#define KOKKOSBLAS_TRTRI_PERF_TEST_H_
+
+//#include <complex.h>
+#include "KokkosBlas_common.hpp"
+
+#include <Kokkos_Random.hpp>
+
+#include <KokkosBlas_trtri.hpp>
+
+#include "KokkosBatched_Trtri_Decl.hpp"
+#include "KokkosBatched_Trtri_Serial_Impl.hpp"
+#include "KokkosBatched_Util.hpp"
+
+//#define TRTRI_PERF_TEST_DEBUG
+
+// Forward declarations
+void do_trtri_serial_blas(options_t options);
+void do_trtri_serial_batched(options_t options);
+void do_trtri_parallel_blas(options_t options);
```

Comment:
@e10harvey I am just confused, please remind me: what do ```do_trtri_serial_blas``` and ```do_trtri_parallel_blas``` differ?

Reply:
`do_trtri_serial_blas` times the invocation of `KokkosBlas::trtri` from a for-loop.

`do_trtri_parallel_blas` times the invocation of `KokkosBlas::trtri` from a kokkos parallel_for-loop.

Reply:
@e10harvey Are performance tests also aimed for TPL (cuBLAS, MAGMA)? For example,  can we test the invocation of ```KokkosBlas::trtri``` calling MAGMA ```trtri``` from a kokkos parallel_for-loop?

Reply:
@vqd8a: Yes, they are aimed for TPLs as well. Attempting to invoke `KokkosBlas::trtri` calling MAGMA trtri from a kokkos parallel_for-loop will produce a runtime error.

Reply:
@e10harvey I just feel like ```do_trtri_parallel_blas``` is to test batched operation. Sorry, maybe I missed something. What does ```do_trtri_serial_batched``` do?

Reply:
@vqd8a: No worries. `do_trtri_serial_batched` times the invocation of `KokkosBatched::trtri` from a for-loop.

Reply:
Thanks, @e10harvey .